cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4MeasureService.java

```diff
@@ -57,6 +62,65 @@ public MeasureReport evaluate(
             String productLine,
             String practitioner) {
 
+        return evaluateMeasureCaptureDefs(
+                        measure,
+                        periodStart,
+                        periodEnd,
+                        reportType,
+                        subjectId,
+                        lastReceivedOn,
+                        contentEndpoint,
+                        terminologyEndpoint,
+                        dataEndpoint,
+                        additionalData,
+                        parameters,
+                        productLine,
+                        practitioner)
+                .measureReport();
+    }
+
+    /**
+     * Test-visible evaluation method that captures both MeasureDef and MeasureReport.
+     * <p>
+     * <strong>TEST INFRASTRUCTURE ONLY - DO NOT USE IN PRODUCTION CODE</strong>
+     * </p>
+     * <p>
+     * This method is package-private and annotated with @VisibleForTesting to support
+     * test frameworks that need to assert on both pre-scoring state (MeasureDef) and
+     * post-scoring state (MeasureReport).
+     * </p>
+     *
+     * @param measure Either canonical URL, ID, or Measure resource
+     * @param periodStart start date of Measurement Period
+     * @param periodEnd end date of Measurement Period
+     * @param reportType type of report
+     * @param subjectId the subject ID
+     * @param lastReceivedOn last received on date
+     * @param contentEndpoint content endpoint
+     * @param terminologyEndpoint terminology endpoint
+     * @param dataEndpoint data endpoint
+     * @param additionalData additional data bundle
+     * @param parameters CQL parameters
+     * @param productLine product line
+     * @param practitioner practitioner ID
+     * @return MeasureDefAndR4MeasureReport containing both MeasureDef and MeasureReport
+     */
+    @VisibleForTesting
+    MeasureDefAndR4MeasureReport evaluateMeasureCaptureDefs(
+            Either3<CanonicalType, IdType, Measure> measure,
+            @Nullable ZonedDateTime periodStart,
+            @Nullable ZonedDateTime periodEnd,
+            String reportType,
+            String subjectId,
+            String lastReceivedOn,
+            Endpoint contentEndpoint,
+            Endpoint terminologyEndpoint,
+            Endpoint dataEndpoint,
+            Bundle additionalData,
+            Parameters parameters,
+            String productLine,
+            String practitioner) {
```

Comment:
thought: I see... seems like this is a case of suboptimal layering, right? A MeasureDef ought be to be able to be evaluated on its own, without any reference to a FHIR measure.

Reply:
That's the ultimate goal:  I'd like to tackle this in a separate PR, since Justin asked me to do invasive changes to my original design, so I'd like to get this merged.